src/components/AddDatasetToCollection.vue

```diff
@@ -0,0 +1,371 @@
+            ref="configuration"
+            :datasetId="option.uploadedDatasetId"
+            :autoDatasetRoute="false"
+            @log="option.configurationLogs = $event"
+            @generatedJson="addDatasetConfigurationDone"
+            :class="{ 'd-none': !option.editVariables }"
+          />
+          <div class="title mb-2">
+            Configuring the dataset
+          </div>
+          <v-progress-circular indeterminate />
+          <div class="code-container" v-if="option.configurationLogs">
+            <code class="code-block">
+              {{ option.configurationLogs }}
+            </code>
+          </div>
+        </template>
+      </template>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from "vue-property-decorator";
+
+import { getDatasetCompatibility } from "@/store/GirderAPI";
+import { IDataset, IDatasetConfiguration, areCompatibles } from "@/store/model";
+import girderResources from "@/store/girderResources";
+import store from "@/store";
+
+import { isDatasetFolder } from "@/utils/girderSelectable";
+import { logError } from "@/utils/log";
+import { IGirderSelectAble } from "@/girder";
+
+import CustomFileManager from "@/components/CustomFileManager.vue";
+import MultiSourceConfiguration from "@/views/dataset/MultiSourceConfiguration.vue";
+import NewDataset from "@/views/dataset/NewDataset.vue";
+
+type TAddDatasetOption =
+  | {
+      type: "upload";
+      editVariables: boolean;
+      configuring: boolean;
+      configurationLogs: string;
+      uploadedDatasetId: string | null;
+    }
+  | {
+      type: "add";
+      datasets: IDataset[];
+      warnings: string[];
+      // Enable navigation in the component
+      datasetSelectLocation: IGirderSelectAble | null;
+    };
+
+function defaultDatasetUploadOption(): TAddDatasetOption {
+  return {
+    type: "upload",
+    editVariables: false,
+    configuring: false,
+    configurationLogs: "",
+    uploadedDatasetId: null
+  };
+}
+
+function defaultDatasetAddOption(): TAddDatasetOption {
+  return {
+    type: "add",
+    datasets: [],
+    warnings: [],
+    datasetSelectLocation: null
+  };
+}
+
+@Component({
+  components: { CustomFileManager, MultiSourceConfiguration, NewDataset }
+})
+export default class AddDatasetToCollection extends Vue {
+  readonly girderResources = girderResources;
+  readonly store = store;
+
+  @Prop({ required: true })
+  collection!: IDatasetConfiguration;
+
+  option: TAddDatasetOption = defaultDatasetUploadOption();
+
+  $refs!: {
+    configuration?: MultiSourceConfiguration;
+  };
+
+  get addDatasetOptionType(): TAddDatasetOption["type"] {
+    return this.option.type;
+  }
+
+  set addDatasetOptionType(type: TAddDatasetOption["type"]) {
+    switch (type) {
+      case "add":
+        this.option = defaultDatasetAddOption();
+        break;
+      case "upload":
+        this.option = defaultDatasetUploadOption();
+        break;
+    }
```

Comment:
no default case ?

Reply:
The two possible values for `type` are checked
TypeScript agrees with it and does not ask for more
I can add a `default:` above `case "upload":` but it should never be used

Reply:
sounds good, just wanted to make sure there is no possible troubles here